File: backend/app/api/schemas/conversation.py
"""HTTP response schema for persisted conversations."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.api.schemas.message import MessageResponse


class ConversationResponse(BaseModel):
    """Serialized persisted conversation."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    session_id: str
    title: str | None
    created_at: datetime
    updated_at: datetime
    messages: list[MessageResponse]

File: frontend/src/schemas/conversation.ts
import { z } from "zod";

import { messageSchema } from "@/schemas/message";

export const conversationSchema = z.object({
  id: z.uuid(),
  session_id: z.string(),
  title: z.string().nullable(),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
  messages: z.array(messageSchema),
});

export const conversationsSchema = z.array(conversationSchema);
